back/engines/commercial/machine_translations/app/controllers/machine_translations/web_api/v1/machine_translations_controller.rb

```diff
@@ -32,6 +32,12 @@ def show
             begin
               @translation = MachineTranslationService.new.build_translation_for @translation_attributes
               authorize @translation
+            
+            # Occasional SQL-injection attacks have been attempted via values for attribute_name,
+            # in which case we ensure @translation == nil, and rescue the resulting Pundit error.
+            rescue Pundit::NotDefinedError
+              render json: { errors: { base: [{ error: 'unable_to_translate' }] } }, status: :unprocessable_entity
```

Comment:
Not sure this is the best error message, but I wanted to keep it vague (avoid providing more information than necessary) since it might be a response to an attempted SQL-injection attack.